src/utils/compacted-merge.ts

```diff
@@ -0,0 +1,16 @@
+import omitBy from 'lodash/omitBy'
+import isNil from 'lodash/isNil'
+
+// A typical ES6 destructuring merge ({ ...obj1, ...obj2 }) will allow values
+// of `null` and `undefined` to override actual values. This compactedMerge
+// function first removes the `null` and `undefined` key-value pairs from each
+// object before merging them.
+//
+// This ensures that if the first object has a key-value pair like
+// `lesson_view_url: '/some/url'` that it doesn't get overridden by the second
+// object having `lesson_view_url: undefined`.
+const compactedMerge = (obj1: object, obj2: object): object => {
+  return {...omitBy(obj1, isNil), ...omitBy(obj2, isNil)}
+}
```

Comment:
`pickBy` is my goto for this. Under the hood it uses `identity` and is very effective for removing `null` and `undefined` 
![Screen Shot 2022-08-12 at 09 26 39](https://user-images.githubusercontent.com/86834/184402274-5ec32f2d-08ae-4e04-b83f-6922665c295d.png)

Reply:
oh nice, didn't know about `pickBy`. That's a more direct way to do what I'm doing here. I'll update it.

Reply:
this is a problem with graphql as well, where `null` the devil shows up

`null` is an object and super weird

![image](https://user-images.githubusercontent.com/86834/184411193-31ebb5a2-b61d-4136-9ff9-07a890f6cf0b.png)